pkg/api/vlabs/types.go

```diff
@@ -332,10 +332,12 @@ type DcosConfig struct {
 type OpenShiftConfig struct {
 	KubernetesConfig *KubernetesConfig `json:"kubernetesConfig,omitempty"`
 
-	// ClusterUsername and ClusterPassword are temporary before AAD
-	// authentication is enabled, and will be removed subsequently.
+	// ClusterUsername and ClusterPassword are temporary, do not rely on them.
```

Comment:
How about a top-level struct dedicated to authentication?

Reply:
Will be talking about this at the next arch call, the first thought was a slimmed down usage of our IdentityProviders in order to keep  the configuration space the same (or nearly) as a normal master config but that had some complications that @jim-minter will explain